server/src/main/java/com/cloud/agent/manager/allocator/impl/TestingAllocator.java

```diff
@@ -28,32 +28,24 @@
 import com.cloud.host.Host;
 import com.cloud.host.Host.Type;
 import com.cloud.host.dao.HostDao;
-import com.cloud.offering.ServiceOffering;
 import com.cloud.utils.component.AdapterBase;
-import com.cloud.vm.VirtualMachine;
 import com.cloud.vm.VirtualMachineProfile;
 
 public class TestingAllocator extends AdapterBase implements HostAllocator {
```

Comment:
why is this not a BaseAllocator-child, @BryanMLima ?

Reply:
@DaanHoogland, because `TestingAllocator` does not have any methods in common with `BaseAllocator`. Is there a reason that it should extend it?

Reply:
No, but it seems an outlier and the name also suggests it is intended to be a mock version for testing what would make less sense if it is not based on the same.  I'd have to study it's use to give a definite answer.